event_model/__init__.py

```diff
@@ -1119,7 +1135,17 @@ def __init__(self, key, message):
 
 
 class DuplicateHandler(EventModelRuntimeError):
-    "raised when a handler is already registered for a given spec"
+    """raised when a handler is already registered for a given spec"""
+    ...
+
+
+class InvalidData(EventModelError):
+    """raised when the data is invalid"""
+    ...
+
+
+class MismatchedDataKeys(InvalidData):
+    """raised when event['data'].keys() != descriptor['data_keys'].keys()"""
```

Comment:
could also be when any of the data keys structures, which include `event['timestamps'].keys()` and `event['filled'].keys()`, are mutually out of sync